java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -58,17 +61,20 @@ class EntityHandleManager(
         storageKey: StorageKey,
         schema: Schema,
         idGenerator: Id.Generator = Id.Generator.newSession()
-    ): SingletonHandleAdapter<T> {
-        val storageHandle = handleManager.rawEntitySingletonHandle(
-            storageKey,
-            schema,
-            canRead = mode.canRead,
-            name = idGenerator.newChildId(
-                idGenerator.newChildId(arcId.toArcId(), hostId),
-                name
-            ).toString()
-        )
-        return SingletonHandleAdapter(mode, name, entitySpec, storageHandle, idGenerator)
+    ) = handleManager.rawEntitySingletonHandle(
+        storageKey,
+        schema,
+        name = idGenerator.newChildId(
+            idGenerator.newChildId(arcId.toArcId(), hostId),
+            name
+        ).toString()
```

Comment:
You could define a helper method for this, would be slightly clearer.

Reply:
I think the `ensureIdentified` helper needs to be modified to use the generated `arcId` as now, so I will do some IdGenerator cleanup in a subsequent PR.